### DISCUSSION TITLE
<Pushing projects to github>

### DISCUSSION THREAD

**Discussion by Patrickodida** [community/community#67632] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

In an attempt to push my project, am getting the following error. please i need help. thank you!
git version 2.42.0.windows.2
PS C:\Users\User\Documents\Odida\Calculator> pwd

Path
----
C:\Users\User\Documents\Odida\Calculator

PS C:\Users\User\Documents\Odida\Calculator> git init
Initialized empty Git repository in C:/Users/User/Documents/Odida/Calculator/.git/
PS C:\Users\User\Documents\Odida\Calculator> git status
On branch master

No commits yet

Changes to be committed:
        new file:   main.css
        new file:   script.js
PS C:\Users\User\Documents\Odida\Calculator> git commit -m "First Commit"
 3 files changed, 148 insertions(+)
 create mode 100644 index.html
 create mode 100644 main.css
 create mode 100644 script.js
PS C:\Users\User\Documents\Odida\Calculator> git remote add origin https://github.com/Patrickodida/JavaScript-Projects.git
PS C:\Users\User\Documents\Odida\Calculator> git push origin master
remote: Permission to Patrickodida/JavaScript-Projects.git denied to StaceySamoe.
fatal: unable to access 'https://github.com/Patrickodida/JavaScript-Projects.git/': The requested URL returned error: 403PS C:\Users\User\Documents\Odida\Calculator>

**Comment by seyLu** [comment]
#### Hi @Patrickodida,

Check your git credentials:
> remote: Permission to Patrickodida/JavaScript-Projects.git denied to StaceySamoe.

```bash
git config --global --list
# check user.name and user.email if it matches your github profile
``` 

If it does, then make sure your local machine is authenticated to access your Github account. You can do this via PAT or SSH (if it's your own machine, then it is better to use SSH).

You can read more on:
- [Git Docs: First-time Git setup](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup)
- [Github Docs: Authentication to Github](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/about-authentication-to-github)

**Reply by Patrickodida** [reply]
Thank you seyLu!, I have managed to push.

**Comment by 1amkaiz3n** [comment]
The error message you're encountering (error: 403 Forbidden) indicates that you don't have permission to push to the repository https://github.com/Patrickodida/JavaScript-Projects.git. This could be due to a couple of reasons:

1.Authentication Issue:
Make sure you are authenticated with the correct GitHub account. If you're trying to push to Patrickodida/JavaScript-Projects.git, you need to be logged in as the user Patrickodida on your local Git.

2.Permission Issue:
Ensure that the GitHub account you are using (Patrickodida) has the necessary permissions to push to the repository. You need to have write access to the repository to push changes.

**To resolve this issue:**

- Make sure you are logged into the correct GitHub account using git config --global user.name and git config --global user.email.

- Double-check that you have the correct permissions to push to the repository. If the repository is owned by a different user or organization, you may need to fork the repository and push to your fork.

- If you've forked the repository, make sure you are pushing to your fork (https://github.com/YourUsername/JavaScript-Projects.git).

- If the repository is private, make sure you have the necessary access rights or have been added as a collaborator.

If you continue to have issues, ensure you're using the correct GitHub account and that the account has the necessary permissions to push to the repository. If you're still facing problems, consider reaching out to the repository owner for assistance.

**Comment by Dawit0101** [comment]
To resolve this issue, you can follow these steps:

1. Ensure that you are logged in with the correct GitHub account:

- Open a terminal.

- Use the git config --global user.name and git config --global user.email commands to verify that your Git username and email are correctly set to the GitHub account you want to use.

2. If the username and email are incorrect, you can set them using these commands:
`git config --global user.name "Your GitHub Username"
git config --global user.email "Your GitHub Email"`

3. After verifying that you are using the correct GitHub account, try pushing again with the following command:
`git push origin master`

if my methods will fix your issues, don't forget to follow my account.
